src/python/pants/backend/python/lint/flake8/subsystem.py

```diff
@@ -92,6 +109,25 @@ def register_options(cls, register):
                 f"non-standard location."
             ),
         )
+        register(
+            "--source-plugins",
+            type=list,
+            member_type=target_option,
+            advanced=True,
+            help=(
+                "An optional list of `python_sources` target addresses to load first-party "
+                "plugins.\n\nYou must set the plugin's parent directory as a source root. For "
+                "example, if your plugin is at `build-support/flake8/custom_plugin.py`, add "
+                "'build-support/flake8' to `[source].root_patterns` in `pants.toml`. This is "
+                "necessary for Pants to know how to tell Flake8 to discover your plugin. See "
+                f"{doc_url('source-roots')}\n\nYou must also set `[flake8:local-plugins]` in "
+                "your Flake8 config file."
+                "\n\nWhile your plugin's code can depend on other first-party code and third-party "
+                "requirements, all first-party dependencies of the plugin must live in the same "
+                "directory or a subdirectory.\n\nTo instead load third-party plugins, set the "
```

Comment:
Is this restriction true for Flake8? It is not true for MyPy.

Reply:
It's a reasonable restriction imo. Probably makes `EXTRA_SYS_PATH` easier to deal with - it's only at most the plugins prefix folder. So I think this is fine, even if it possibly is not technically necessary.

Reply:
It might warrant a test :thinking: 
Do you have an example for `mypy` we can use as a template?

Reply:
https://github.com/pantsbuild/pants/blob/1027041d129dd2387b5be4ee32d06dcfb70b01d3/src/python/pants/backend/python/typecheck/mypy/rules_integration_test.py#L533-L640

But I wouldn't spend too much time on this. Imo it's totally reasonable to at least for now not let you import from any arbitrary part of the codebase. Not worth the code (including test code) getting even more complex.

Reply:
I'll punt for now then. Can be revisited later :)